src/Common/src/System/Net/Security/CertificateValidation.Unix.cs

```diff
@@ -45,6 +45,11 @@ internal static SslPolicyErrors BuildChainAndVerifyProperties(X509Chain chain, X
                     // like "all characters being within [a-z0-9.-]+"
                     string matchName = s_idnMapping.GetAscii(hostName);
                     hostNameMatch = Interop.Crypto.CheckX509Hostname(certHandle, matchName, matchName.Length);
+
+                    if (hostNameMatch < 0)
```

Comment:
since you've added the check, what are the cases where this doesn't fail but we return `errors | SslPolicyErrors.RemoteCertificateNameMismatch`

Reply:
I'm not sure I follow.  NameMismatch is reported when the error is 0.  If it's less than 0 that's an exception, if it's greater than 0 it should be 1.